node/src/neighborhood/mod.rs

```diff
@@ -1288,18 +1481,21 @@ impl Neighborhood {
     #[allow(clippy::too_many_arguments)]
     fn routing_engine<'a>(
         &'a self,
-        prefix: Vec<&'a PublicKey>,
+        prefix: &mut Vec<&'a PublicKey>,
         undesirability: i64,
         target_opt: Option<&'a PublicKey>,
         hops_remaining: usize,
         payload_size: usize,
         direction: RouteDirection,
         minimum_undesirability: &mut i64,
         hostname_opt: Option<&str>,
+        research_neighborhood: bool,
+        research_exits: &mut Vec<&'a PublicKey>,
```

Comment:
Jinak opet tak trochu lingvisticky problem. 

Z meho chapani se pouziti slovosledu `research_exits` sem moc nehodi. V prekladu je to vlastne rozkaz, ktery se v programovani nejcasteji pouziva budto pro znazorneni podminky (a proto by clovek mohl cekat boolean), anebo druhy pripad, jenz by odkazoval na funkci (Tim rikas, jakmile tenhle argument pouzijes - pomoci volani - tak se vykona to a to). Jenze ty neresis ani jeden z tech dvou pripadu. Proto si myslim, je ten nazev nepatricny.  Spravneji by to asi melo byt neco jako: `researched_exits`,  `discovered_exits` nebo pripadne i `exits_research`, coz se mi taky nezamlouva, ale stale je to lepsi nez ta tva varianta, nebot je to alespon podstatne jmeno.